src/oc/auth/async/slack_router.clj

```diff
@@ -0,0 +1,124 @@
+(ns oc.auth.async.slack-router
+  "Consume slack messages from SQS. This SQS queue is subscribed to the Slack
+   message SNS topic.
+  "
+  (:require
+   [clojure.core.async :as async :refer (<!! >!!)]
+   [cheshire.core :as json]
+   [oc.lib.sqs :as sqs]
+   [oc.lib.db.pool :as pool]
+   [oc.auth.config :as config]
+   [oc.auth.resources.team :as team-res]
+   [oc.auth.resources.slack-org :as slack-res]
+   [taoensso.timbre :as timbre]))
+
+;; ----- core.async -----
+
+(defonce slack-router-chan (async/chan 10000)) ; buffered channel
+
+(defonce slack-router-go (atom nil))
+
+(defn slack-event
+  "
+  Handle a message event from Slack. Ignore events that aren't threaded, or that are from us.
+  Idea here is to do very minimal processing and get a 200 back to Slack as fast as possible as this is a 'fire hose'
+  of requests. So minimal logging and minimal handling of the request.
+  Message events look like:
+  {'token' 'IxT9ZaxvjqRdKxYtWdTw21Xv',
+   'team_id' 'T06SBMH60',
+   'api_app_id' 'A0CHN2UDB',
+   'event' {'type' 'message',
+            'user' 'U06SBTXJR',
+            'text' 'Call me back here',
+            'thread_ts' '1494262410.072574',
+            'parent_user_id' 'U06SBTXJR',
+            'ts' '1494281750.011785',
+            'channel' 'C10A1P4H2',
+            'event_ts' '1494281750.011785'},
+    'type' 'event_callback',
+    'authed_users' ['U06SBTXJR'],
+    'event_id' 'Ev5B8YSYQ6',
+    'event_time' 1494281750}
+  "
+  [db-pool body]
+  (let [type (:type body)
+        token (:token body)
+        event (:event body)
+        channel (:channel event)
+        thread (:thread_ts event)]
+    ;; Token check
+    (if-not (= token config/slack-verification-token)
```

Comment:
@thepug i don't have any `slack-verification-token` set for localhost, where can i find it?

Reply:
`export OPEN_COMPANY_SLACK_VERIFICATION_TOKEN="xxxx"` is in my environment.